### ISSUE TITLE
Thumbnails werden nicht angezeigt 

### ISSUE THREAD

**Issue by markwaypoint** [weg-li/weg-li#814]
Ich benutze weg.li mit iOS Safari, nach dem Hochladen werden ab dem Zuordungsdialog vermehrt fehlende Grafiken statt der Thumbnails angezeigt, die Zuordnung verkompliziert dieses. 
![Uploading IMG_4121.jpeg…]()

**Comment by phoet** [comment]
Das ist ja vielsagend, dass der Upload hier bei GitHub auch nicht funktioniert hat 😄

**Comment by phoet** [comment]
Ich kann so schon sagen, dass es sich hier höchstwahrscheinlich um irgend einen mist von Apple handelt, da nur dort dieses Problem aufzutreten scheint. Da ich keine Möglichkeit habe es zu debuggen, wird sich da vorerst auch nichts dran ändern.

**Comment by phoet** [comment]
Irgendwo gab es schon eine ähnliche Fehlermeldung, konnte ich aber leider nicht mehr wiederfinden.

**Comment by markwaypoint** [comment]
Ist eindeutig Safari + IOS Problem, Win11 + Firefox haben mit den gleichen Uploads keine Probleme, konnte ich nicht früher ausprobieren. Kannst du schließen.

**Comment by plantzilla** [comment]
Hallo, das Problem tritt sowohl unter iOS, iPadOS und macOS, jeweils in den aktuellsten Versionen, auf. Es werden manche wenige Thumbnails jedoch geladen. Unter allen aktuellen Apple-Geräte ist weg.li dadurch kaum noch nutzbar, vor allem auf mobilen Geräten, da die ja keine alternative Browser-Engine zulassen bzw. es auch noch keine anderen gibt. Ich habe das auf macOS nachvollziehen können, hier eine Beispiel-URL, die nicht angezeigt wird:
https://images.weg.li/cdn-cgi/image/width=200,height=200,fit=scale-down,metadata=keep,quality=90/storage/aoaix1urrc93c39673hwzfx49t2g.jpg
Diese wird auch bei direktem Aufruf nur mit einem ?-Symbol angezeigt.
<img width="549" alt="Bildschirmfoto 2024-12-28 um 15 30 55" src="https://github.com/user-attachments/assets/6e98b94e-4e7d-4e58-9770-1e249f1872b6" />
Mit den Entwicklertools steht dann folgendes in der Konsole:
<img width="1229" alt="Bildschirmfoto 2024-12-28 um 15 33 17" src="https://github.com/user-attachments/assets/06f7f293-10f4-4860-9ade-492c04230169" />
Hier das zugehörige HTML-Element beim Aufruf einer Meldung:
`<img class="item-photo" src="https://images.weg.li/cdn-cgi/image/width=200,height=200,fit=scale-down,metadata=keep,quality=90/storage/pwlh8f375ipxqlvo4skva59jttik.jpg">`
Die große Version wird problemlos angezeigt:
`https://images.weg.li/cdn-cgi/image/width=1280,height=1280,fit=scale-down,metadata=keep,quality=90/storage/pwlh8f375ipxqlvo4skva59jttik.jpg`
Wenn man manuell mal ein paar Werte ausprobiert, dann klappt es mit Werten von width/height KLEINER als 1009 nicht mehr.
Ein bisschen weitere Spielerei deckt noch diesen Fehler auf:
<img width="422" alt="Bildschirmfoto 2024-12-28 um 15 55 22" src="https://github.com/user-attachments/assets/dd9fdd61-d4f7-41fb-9629-777b282c68b6" />
Diese Content Policy steht im HTTP Antwort Header:
<img width="521" alt="Bildschirmfoto 2024-12-28 um 15 56 11" src="https://github.com/user-attachments/assets/45fc0465-045a-4d77-9ebe-1c0ca9f9d9e7" />
Im Firefox läuft diese übrigens nur auf eine Warnung, und nicht auf einen Fehler:
<img width="696" alt="Bildschirmfoto 2024-12-28 um 15 57 28" src="https://github.com/user-attachments/assets/9d1984f0-ca0a-4e44-aee7-c8a834ca7c7d" />

Leider gibt Safari diese Fehler nur aus, wenn man das Bild über die URL wie oben direkt aufruft. Bei normaler Websitebenutzung werden gar keine Fehler angezeigt. Beim entsprechenden Thumbnail wird unter "Quellen" nur angezeigt: "Ressource hat keinen Inhalt". Ich habe leider gerade keine Idee, wie man dem weiter auf den Grund gehen kann. Internetsuche hat leider nichts richtungsweisendes zutage gebracht.

**Comment by plantzilla** [comment]
Die "kaputten" Bilder verursachen auch lokal unter macOS Probleme. Der Finder zeigt keine Vorschau an und auch die größere Finder-Vorschau (Leertaste) zeigt nichts an. Die App "Vorschau" öffnet das Bild, kann es aber nicht wieder speichern oder bearbeiten (natürlich ohne sinnvolle Fehlermeldung...). Ich vermute daher, dass der Cloudflare Image Resizer irgendetwas mit den Bilddaten macht, womit die native macOS JPEG-Bibliothek nicht klarkommt. Verschiedene andere Tools zeigen keine Fehler mit den skalierten Bildern.
Hierzu habe ich diesen Artikel gefunden, der das Problem korrekt beschreibt inkl. Lösung:
https://decovar.dev/blog/2019/01/03/fixing-corrupted-jpg-on-macos/
Das habe ich nachgestellt und kann es bestätigen. Ändert man nur diese zwei Bytes in der Datei, wird sie auch in Safari wieder korrekt angezeigt. Es handelt sich also gar nicht um ein Problem mit Safari an sich, sondern Cloudflare oder macOS bzw. seinem Unterbau, der ja sicherlich weitgehend mit iOS etc. gleichzieht. Wer da jetzt nicht JPEG-spezifikationskonform arbeitet, erschließt sich mir aber nicht.

**Comment by plantzilla** [comment]
Ein einfacher Fix wäre, den Aufruf statt mit ´metadata=keep´ mit ´metadata=none´ laufen zu lassen für alle angezeigten Bilder. Damit geht es nämlich. Die Download-Schaltfläche kann dann ja was anderes tun.

**Comment by phoet** [comment]
Danke für das intensive debugging, schaue ich mir genauer an. Grundsätzlich lässt sich sagen, dass dies entweder mit einem neuen iOS oder Safari release eingeführt worden ist. Ich würde es (mal wieder) als einen Apple Bug klassifizieren, wie schon so oft in der Vergangenheit.

**Comment by plantzilla** [comment]
Danke, dass du es dir anschaust! Ich bin froh, dass es dieses Projekt gibt. Mag sein, dass es ein Apple Bug ist - der verlinkte Post ist aber schon von 2019 und es lief ja rund, bis vor ein paar Monaten. Vielleicht hat ja auch Cloudflare an seinem Dienst geschraubt...

**Comment by phoet** [comment]
Nee, wir haben regressionstests, dass die Bilder identisch bleiben.

**Comment by phoet** [comment]
Gleich geht die änderung von `keep` zu `none` live, danach gerne einmal testen ob es geholfen hat und feedback geben. Wenn es nicht hilft ändere ich es wieder zurück, da wir sonst den gesammten bilder-cache verlieren.

**Comment by plantzilla** [comment]
Bei mir wird noch mit `keep` ausgeliefert.

**Comment by plantzilla** [comment]
Hurra! Alle Falschparker sind wieder zu sehen. Getestet auf macOS 15.2/Safari, iOS 18.2/Safari, iPadOS 18.2/Safari, Windows 11/Firefox.
<img width="273" alt="Bildschirmfoto 2024-12-29 um 13 51 21" src="https://github.com/user-attachments/assets/7ffa2f3b-8d01-4751-af8f-ae268684c883" />

**Comment by mnin** [comment]
@phoet ist mir nicht bekannt.
